designsafe/apps/api/datafiles/operations/tapis_operations.py

```diff
@@ -43,15 +44,63 @@ def listing(client, system, path, offset=0, limit=100, q=None, *args, **kwargs):
 
     if q:
         return search(client, system, path, offset=0, limit=100, query_string=q, **kwargs)
+    if system == settings.PUBLISHED_SYSTEM:
+        raw_listing = client.files.listFiles(
+            systemId=system,
+            path=(path or '/'),
+            offset=int(offset),
+            limit=int(limit),
+            headers={"X-Tapis-Tracking-ID": kwargs.get("tapis_tracking_id", "")}
+        )
+        listing = []
+        for f in raw_listing:
+            type_str = 'dir' if f.type == 'dir' else 'file'
+            format_str = 'folder' if f.type == 'dir' else 'raw'
+            print("Item:", f.name, "type:", getattr(f, "type", None), "path:", f.path)
+
+            if getattr(f, "type", None) == "symbolic_link":
+                accessor = f"tapis://{system}/{f.path}"
+                try:
+                    rec = PublicationSymlink.objects.get(tapis_accessor=accessor)
+                    type_str = rec.type    # 'file' or 'dir'
+                    format_str = 'folder' if type_str == 'dir' else 'raw'
+                    print(f"Symlink override: {accessor} -> {type_str}")
+                except PublicationSymlink.DoesNotExist:
+                    # Fallback
+                    type_str = 'file'
+                    format_str = 'raw'
+                    print(f"Symlink not found in DB, defaulting to file: {accessor}")
+
+            listing.append({
```

Comment:
It's possible we might want to support symlinks in places that aren't published projects. Instead of duplicating all the listing/formatting code inside the conditional block, let's check the `raw_listing` that's already defined in the function (line 90 in this diff) and update its `type` attirbutes if we find a match in the symlink db.